### PR TITLE
Added support to specify the package/folder in the virtual file system for Tiberian Sun Cabal/EVA voices and Nod/GDI icons

### LINKED ISSUE #8803: Remap TS refinery/silo icon color to player color
Tiberian Sun uses the same production icon with different "player" color (Yellow for GDI, Red for Nod) for several structures ([source](http://xhp.xwis.net/ts_icons/)):
 - Refinery
 - Silo
 - EMP Cannon

Since that is a bit restrictive, and we have the technology, I propose we remap those to the actual color the player is.
Those aren't building decoration colors. Those parts of the buildings are remapped to the player color. So why not remap them on the icons as well?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/24986082-remap-ts-refinery-silo-icon-color-to-player-color?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/6353. 
Closes #8803. See http://xhp.xwis.net/ts_icons/ for reference.

### CODE REVIEW

File: OpenRA.Mods.Cnc/CncLoadScreen.cs [hunk 1]
```diff
@@ -41,7 +43,9 @@ public override void Init(Manifest m, Dictionary<string, string> info)
 			r = Game.Renderer;
 			if (r == null) return;
 
-			sheet = new Sheet(Platform.ResolvePath(loadInfo["Image"]));
+			var image = Platform.ResolvePath(info["Image"]);
+			GlobalFileSystem.Mount(Path.GetDirectoryName(image));
```
Comment (pchote) [inline OpenRA.Mods.Cnc/CncLoadScreen.cs]: This looks like a workaround / bandaid for something bogus.  Why are all these additional mounts needed?

Comment (Mailaender) [inline OpenRA.Mods.Cnc/CncLoadScreen.cs]: It is absolute vs. relative path inconsistencies. The game loads and labels a relative one, but the code here wants an absolute one. Never mattered as we simply searched through the whole virtual file system for the one file. This adds redundancy, but saves lookup time.

Comment (pchote) [inline OpenRA.Mods.Cnc/CncLoadScreen.cs]: Changing global state as a side-effect of this otherwise unrelated code is really bad practice, so we'll need to find a better way of solving this problem.

**Comment by Mailaender** [comment]
I went for a package:filename separator that doesn't clash with real folders anymore now.

**Comment by Phrohdoh** [comment]
This works nicely, though I'd like a chance to write code for sequence merging to avoid duplication. :+1:

File: mods/ts/sequences/vehicles.yaml [hunk 2]
```diff
@@ -1,13 +1,21 @@
-mcv:
-	icon: mcvicon
+mcv.gdi:
+	icon: sidec01.mix/mcvicon
```
Comment (GraionDilach) [inline mods/ts/sequences/vehicles.yaml]: You forgot to update this with your new system?

Comment (Mailaender) [inline mods/ts/sequences/vehicles.yaml]: Yep, well spotted. Fixed.

File: mods/ts/sequences/infantry.yaml [hunk 3]
```diff
@@ -1,37 +1,37 @@
-e1:
+e1.gdi:
 	Defaults:
 		Tick: 80
-	stand:
+	stand: e1
```
Comment (GraionDilach) [inline mods/ts/sequences/infantry.yaml]: Why not specifying this -and other infantry filenames - on the Defaults instead of the current override on every subsequence?

File: OpenRA.Game/FileSystem/GlobalFileSystem.cs [hunk 4]
```diff
@@ -185,10 +185,23 @@ public static Stream Open(string filename)
...
+			{
+				var divide = name.Split(':');
+				foldername = divide.First();
+				filename = divide.Last();
+			}
+			else
+				filename = name;
+
+			// Check the cache for a quick lookup if the folder name is unknown
+			if (filename.IndexOfAny(new char[] { '/', '\\' }) == -1 && string.IsNullOrEmpty(foldername))
```
Comment (pchote) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: This forces these faction-specific sprites to take the slow path, which kind of sucks.
The whole filesystem code needs a rewrite, so I won't block the pr over that, but please add a `// TODO` about this.

File: mods/ts/sequences/infantry.yaml (hunk 3 above)
Comment (pchote) [inline mods/ts/sequences/infantry.yaml]: I agree. Can you please fix this?

**Comment by Mailaender** [comment]
Updated.

**Comment by Phrohdoh** [comment]
:+1: works for me. Let's test it in a game?

File: OpenRA.Game/FileSystem/GlobalFileSystem.cs [hunk 5]
```diff
@@ -185,10 +185,23 @@ public static Stream Open(string filename)
...
+			var explicitFolder = name.Contains(':') && !Directory.Exists(Path.GetDirectoryName(name));
+			if (explicitFolder)
+			{
+				var divide = name.Split(':');
+				foldername = divide.First();
+				filename = divide.Last();
+			}
+
+			// Check the cache for a quick lookup if the folder name is unknown
+			// TODO: explicit folders will always take the slow path
```
Comment (pchote) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: I suggest: `// TODO: This disables caching for explicit folder requests` to make this more clear.

File: OpenRA.Game/FileSystem/GlobalFileSystem.cs [hunk 6]
```diff
@@ -214,7 +229,19 @@ public static bool TryOpen(string filename, out Stream s)
 			return false;
 		}
 
-		public static bool Exists(string filename) { return MountedFolders.Any(f => f.Exists(filename)); }
+		public static bool Exists(string name)
+		{
+			var explicitFolder = name.Contains(':');
```
Comment (pchote) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: This will need the same windows safety check as above.

Comment (Mailaender) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: Right. Fixed.

File: OpenRA.Game/FileSystem/GlobalFileSystem.cs (hunk 5 above)
Comment (penev92) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: :+1:

Comment (Mailaender) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: I replaced the `string.IsNullOrEmpty(folder)` with `if (explicitFolder)` which should be faster and essentially say that in code.

Comment (pchote) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: The comment refers to skipping the `GetFromCache` calls inside this block, not anything to do the evaluation of the conditions.

Comment (Mailaender) [inline OpenRA.Game/FileSystem/GlobalFileSystem.cs]: Updated.

**Comment by pchote** [comment]
Code changes look fine to me, and this is a great polish improvement ingame :+1:

**Comment by penev92** [comment]
:+1:

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/5eb464bce4f4040ac6382d066df97d2f649eb97b...9d442967474fbe1c07617b9d67ac85f08992dddc)
